ygot/render.go

```diff
@@ -222,31 +381,42 @@ func sliceToScalarArray(v []interface{}) (*gnmipb.ScalarArray, error) {
 // likely to be suboptimal since it results in very large Notifications for particular
 // structs. There should be some fragmentation of Updates across Notification messages
 // in a future implementation. We return a slice to keep the API stable.
-func leavesToNotifications(leaves map[*path]interface{}, ts int64, prefix []interface{}) ([]*gnmipb.Notification, error) {
+func leavesToNotifications(leaves map[*path]interface{}, ts int64, pfx *pathWrapper) ([]*gnmipb.Notification, error) {
 	n := &gnmipb.Notification{
 		Timestamp: ts,
 	}
 
-	pfx := interfacePathAsgNMIPath(prefix)
-	if len(pfx.Element) > 0 {
-		n.Prefix = pfx
-	}
-
-	for p, v := range leaves {
-		path, err := stripPrefix(p.p, prefix)
+	if pfx != nil {
+		if pfx.pathFormat != ElementPath && pfx.pathFormat != PathElemPath {
+			return nil, fmt.Errorf("invalid path format specified: %v", pfx.pathFormat)
+		}
 
-		for _, p := range path {
-			if p == nil {
-				log.Infof("leavesToNotifications got nil in path: %v", path)
+		switch {
```

Comment:
You could probably omit this switch and always perform both actions i.e. 
n.Prefix = &gnmipb.Path{
￼ Element: pfx.elementPath,
￼ Elem: pfx.structuredPath,
}

Reply:
We can't simply do this, since we'll end up with `n.Prefix.Element == []string{}` rather than `n.Prefix.Element == nil` in the case that we have asked for string slice paths, but we can handle this case specifically.